File: examples/hello.c
// Example C script for testing Ditto
#include <stdio.h>

int main(int argc, char *argv[]) {
    // Basic output
    printf("Hello from C!\n");
    printf("Running on Ditto embedded interpreter\n\n");
    
    // Variables
    int number = 42;
    float pi = 3.14159;
    char *message = "Ditto is awesome";
    
    printf("Number: %d\n", number);
    printf("Pi: %f\n", pi);
    printf("Message: %s\n\n", message);
    
    // Arithmetic
    int a = 10, b = 5;
    printf("Arithmetic:\n");
    printf("  %d + %d = %d\n", a, b, a + b);
    printf("  %d - %d = %d\n", a, b, a - b);
    printf("  %d * %d = %d\n", a, b, a * b);
    printf("  %d / %d = %d\n", a, b, a / b);
    
    // Loop
    printf("\nCounting: ");
    for (int i = 1; i <= 5; i++) {
        printf("%d ", i);
    }
    printf("\n");
    
    return 0;
}

File: examples/hello.lua
-- Example Lua script for testing Ditto
print("Hello from Lua!")
print("Running on Ditto embedded interpreter")

-- Variables
local name = "Ditto"
local version = 0.1

-- String concatenation
io.write("Project: " .. name .. " v" .. version .. "\n")

-- Numbers
local a = 10
local b = 20
print("Sum: " .. (a + b))

-- For loop
print("Counting:")
for i = 1, 5 do
    print("  " .. i)
end

-- Table
local colors = {"red", "green", "blue"}
print("Colors: " .. #colors .. " items")

-- Functions
function greet(person)
    print("Hello, " .. person .. "!")
end

greet("World")
